From 0892d1ba1bec8ee7a49f720cbbce7d595e2b2cff Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Sun, 30 Sep 2018 18:32:55 -0700
Subject: [PATCH 04/13] DO-NOT-UPSTREAM: Disable CPU_BIG_ENDIAN with Clang

This is currently broken and requires disabling it to build anyways.

Link: https://github.com/ClangBuiltLinux/linux/issues/36
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 arch/arm/mm/Kconfig | 2 +-
 arch/arm64/Kconfig  | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mm/Kconfig b/arch/arm/mm/Kconfig
index b169e580bf82..63fdc70552aa 100644
--- a/arch/arm/mm/Kconfig
+++ b/arch/arm/mm/Kconfig
@@ -741,7 +741,7 @@ config SWP_EMULATE
 
 config CPU_BIG_ENDIAN
 	bool "Build big-endian kernel"
-	depends on ARCH_SUPPORTS_BIG_ENDIAN
+	depends on ARCH_SUPPORTS_BIG_ENDIAN && !CC_IS_CLANG
 	help
 	  Say Y if you plan on running a kernel in big-endian mode.
 	  Note that your board must be properly built and your board
diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index a2e27485d68c..4dc233cbfdc1 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -711,6 +711,7 @@ config ARM64_PA_BITS
 
 config CPU_BIG_ENDIAN
        bool "Build big-endian kernel"
+       depends on !CC_IS_CLANG
        help
          Say Y if you plan on running a kernel in big-endian mode.
 
-- 
2.19.0

